backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/lower/InteropLowering.kt

```diff
@@ -626,7 +676,7 @@ internal class InteropLoweringPart1(val context: Context) : IrBuildingTransforme
 
             return builder.at(expression).run {
                 val classPtr = getRawPtr(expression.extensionReceiver!!)
-                callAllocAndInit(classPtr, initMethodInfo, arguments)
+                callAllocAndInit(classPtr, initMethodInfo, arguments, expression, callee as IrSimpleFunction)
```

Comment:
Cannot this cast fail?

Reply:
No, because this code is executed only when `callee` is factory method, which is always a (simple) function.